dask/dataframe/io/tests/test_csv.py

```diff
@@ -1083,3 +1083,43 @@ def test_to_csv_paths():
     assert ddf.to_csv("foo*.csv") == ['foo0.csv', 'foo1.csv']
     os.remove('foo0.csv')
     os.remove('foo1.csv')
+
+
+def test_to_csv_header():
+    partition_count = 3
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd', 'e', 'f'],
+                       'y': [1, 2, 3, 4, 5, 6]})
+    ddf = dd.from_pandas(df, npartitions=partition_count)
+
+    # Test normal header case (header=True, header_first_chunk not passed)
```

Comment:
Maybe use `with tmpdir() as dn:`, then you don't have to worry about cleaning up files (which won't be cleaned up anyway if there's an error / failure)